src/Futhark/CLI/Autotune.hs

```diff
@@ -207,80 +197,107 @@ thresholdForest prog = do
 
 --- Doing the atual tuning
 
-intersectRanges :: [(Int, Int)] -> (Int, Int)
-intersectRanges = foldl' f (thresholdMin, thresholdMax)
-  where f (xmin, xmax) (ymin, ymax) =
-          -- XXX: what happens when the intersection is empty?
-          (xmin `max` ymin,
-           xmax `min` ymax)
-
 tuneThreshold :: AutotuneOptions
               -> [(DatasetName, RunDataset, T.Text)]
               -> Path -> (String, Path)
               -> IO Path
-tuneThreshold opts datasets already_tuned (v, v_path) = do
-  ranges <-
-    forM datasets $ \(dataset_name, run, entry_point) ->
-
-    if not $ isPrefixOf (T.unpack entry_point ++ ".") v then do
-        when (optVerbose opts > 0) $
-          putStrLn $ unwords [v, "is irrelevant for", T.unpack entry_point]
-        return (thresholdMin, thresholdMax)
-    else do
-
-      putStrLn $ unwords ["Tuning", v, "on entry point", T.unpack entry_point,
-                          "and dataset", dataset_name]
-
-      sample_run <- run path RunSample
-
-      case sample_run of
-        Left err -> do
-          -- If the sampling run fails, we treat it as zero information.
-          -- One of our ancestor thresholds will have be set such that
-          -- this path is never taken.
-          when (optVerbose opts > 0) $ putStrLn $
-            "Sampling run failed:\n" ++ err
-          return (thresholdMin, thresholdMax)
-        Right (cmps, _) ->
-          case lookup v cmps of
+tuneThreshold opts datasets already_tuned (v, _v_path) = do
+  putStrLn $ unwords ["Tuning threshold", v]
+
+  (e_pars', _) <- getAllEPars already_tuned v
+  let e_pars = sort $ nub $ thresholdMin : thresholdMax : e_pars'
```

Comment:
`nub` is going to destroy us eventually; go through a Set to remove the duplicates.